### ISSUE TITLE
Segfault wiht ghmmp within BRAKER

### ISSUE THREAD

**Issue by tshori** [Gaius-Augustus/BRAKER#834]
Hi Folks,

I am not sure if this is the right forum for this question. If not, can you point me in the right direction? I am running or trying to run BRAKER on Ubuntu 22.04 (Kernel version 6.5.2) and it kept crashing \during the RNAseq part of the analysis (long reads and proteins). The actual issue is that gmhmmp is crashing with segfault. At first, I figured it could be an issue with the kernel version, but I later realized that if I reduce the total sequence length  (total, not per sequence) to about 100Mb it will run, but anything above that crashes. It does not appear to be a memory issue unless it is happening incredibly fast. Usage on TOP remains low throughout the process, including prior to the crash. I tried debug mode, but it does not tell me much. I also figure it could be an issue of too many threads running at the same time, but it seems to be running on a single thread.
I wondered if it could be a license issue, but I tested recently downloaded licenses to no avail. I also tried to run ghmmp2, and I see the same problem.

Any help would be great!

Cheers,

Tiago

**Comment by KatharinaHoff** [comment]
It is unlikely a license issue. Your best source for help with gmhmmp issues will be the genemark development team.

**Comment by tshori** [comment]
I am not sure what the issue is per se, but I think I figured it out. It was my incompetence. I missed the part where you had to switch branches for long reads. I also still had $GENE_MARK_PATH set to GeneMark-ETP, not GeneMark-T. Still, it seems the issue was the using the main branch. Braker will crash after the gmhmmp step if you don't set the right path because the folder structure on GeneMark-ETP appears to be different, but changing branches got me through the ghmmp step. I did not look at the differences between the branches to try to figure out where it goes wrong, but it is an easy fix: just follow the instructions! Silly me. 

T.

**Comment by KatharinaHoff** [comment]
Why don’t you use the isoseq container? That is easier.

Tiago Hori ***@***.***> schrieb am Do. 30. Mai 2024 um 16:26:

**Comment by tshori** [comment]
In all honesty, I installed all parts separately because that allow me to swap different version of GeneMark. I don’t think you can do that without messing with the container. However, now I should try the singularity. Thanks!

**Comment by KatharinaHoff** [comment]
I have two separate container images for this. Not sure whether we are
talking about different things. I am referring to the latest BRAKER
container and the isoseq BRAKER container.

Tiago Hori ***@***.***> schrieb am Do. 30. Mai 2024 um 20:20:

**Comment by tshori** [comment]
Yes. I will try the isoseq container. I didn’t realize that they were slightly different, so I wanted to try to use different versions(for different Kernels) to debug. However, the issue is that I was using the wrong pipeline for isoseq, so I should go back and try the isoseq specific container. 

Thanks!
